src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -25,51 +25,48 @@ public class DiagnosticEventTableStorageRepository : IDiagnosticEventRepository,
         private const int TableCreationMaxRetryCount = 5;
 
         private readonly Timer _flushLogsTimer;
-        private readonly IConfiguration _configuration;
         private readonly IHostIdProvider _hostIdProvider;
         private readonly IEnvironment _environment;
         private readonly ILogger<DiagnosticEventTableStorageRepository> _logger;
+        private readonly IAzureTableStorageProvider _azureTableStorageProvider;
         private readonly IServiceProvider _serviceProvider;
         private readonly object _syncLock = new object();
-
         private ConcurrentDictionary<string, DiagnosticEvent> _events = new ConcurrentDictionary<string, DiagnosticEvent>();
         private TableServiceClient _tableClient;
         private TableClient _diagnosticEventsTable;
         private string _hostId;
         private bool _disposed = false;
         private bool _purged = false;
+        private bool _tableClientInitializationFailed = false;
         private string _tableName;
 
-        internal DiagnosticEventTableStorageRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHostManager,
-            ILogger<DiagnosticEventTableStorageRepository> logger, int logFlushInterval)
+        internal DiagnosticEventTableStorageRepository(IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHostManager,
+            ILogger<DiagnosticEventTableStorageRepository> logger, IAzureTableStorageProvider azureTableStorageProvider, int logFlushInterval)
         {
-            _configuration = configuration;
             _hostIdProvider = hostIdProvider;
             _environment = environment;
             _serviceProvider = scriptHostManager as IServiceProvider;
             _logger = logger;
+            _azureTableStorageProvider = azureTableStorageProvider;
             _flushLogsTimer = new Timer(OnFlushLogs, null, logFlushInterval, logFlushInterval);
         }
 
-        public DiagnosticEventTableStorageRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHost,
-            ILogger<DiagnosticEventTableStorageRepository> logger)
-            : this(configuration, hostIdProvider, environment, scriptHost, logger, LogFlushInterval) { }
+        public DiagnosticEventTableStorageRepository(IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHost, ILogger<DiagnosticEventTableStorageRepository> logger,
+            IAzureTableStorageProvider azureTableStorageProvider)
+            : this(hostIdProvider, environment, scriptHost, logger, azureTableStorageProvider, LogFlushInterval)
+        {
+        }
 
         internal TableServiceClient TableClient
```

Comment:
I know this is not the change from your PR, But I am curious why we have this as a property. Looks like it is being used only in this class. I prefer a private variable for this. (With that, we could hopefully remove the logging from the property.)